Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM golang:1.16 AS builder
+RUN mkdir /app
+ADD . /app
+WORKDIR /app
+RUN CGO_ENABLED=0 GOOS=linux go build -o app cmd/server/main.go
+
+FROM alpine:latest AS production
+COPY --from=builder /app .
+CMD ["./app"]
```

Comment:
The reason for separation is because it is lighter weight.  Without lines 7-8, we would be running with the `GOOS` and that would be bigger than using `alpine` to create a lightweight image to run instead.